pip install error: implicit declaration of function 'unwrap2D' is invalid in C99

## Description
I'm getting an installation error on MacOS using pip

<!--
(Note: for guidance on how to use `scikit-image`, please post instead on https://forum.image.sc/tag/scikit-image)
-->

## Way to reproduce
```python
pip install scikit-image
```


## Version information
```python
# Paste the output of the following python commands
3.9.0 (default, Oct 27 2020, 14:15:17) 
[Clang 12.0.0 (clang-1200.0.32.21)]
macOS-10.15.7-x86_64-i386-64bit
numpy version: 1.19.4
```

skimage/restoration/_unwrap_2d.c:2285:9: error: implicit declaration of function 'unwrap2D' is invalid in C99 [-Werror,-Wimplicit-function-declaration]

See the attached log file the for the full dump:
[scikit_install_log.txt](https://github.com/scikit-image/scikit-image/files/5492160/scikit_install_log.txt)

--- a/skimage/restoration/_unwrap_2d.pyx
+++ b/skimage/restoration/_unwrap_2d.pyx
@@ -11,6 +11,18 @@ cdef extern from *:
                   int wrap_around_x, int wrap_around_y,
                   char use_seed, unsigned int seed) nogil
 
+
+cdef extern from "unwrap_2d_ljmu.h":
+    void unwrap2D(
+            double *wrapped_image,
+            double *UnwrappedImage,
+            unsigned char *input_mask,
+            int image_width, int image_height,
+            int wrap_around_x, int wrap_around_y,
+            char use_seed, unsigned int seed
+            )
+
+
 def unwrap_2d(double[:, ::1] image,
               unsigned char[:, ::1] mask,
               double[:, ::1] unwrapped_image,

--- a/skimage/restoration/_unwrap_2d.pyx
+++ b/skimage/restoration/_unwrap_2d.pyx
@@ -3,14 +3,6 @@
 # cython: nonecheck=False
 # cython: wraparound=False
 
-cdef extern from *:
-    void unwrap2D(double* wrapped_image,
-                  double* unwrapped_image,
-                  unsigned char* input_mask,
-                  int image_width, int image_height,
-                  int wrap_around_x, int wrap_around_y,
-                  char use_seed, unsigned int seed) nogil
-
 
 cdef extern from "unwrap_2d_ljmu.h":
     void unwrap2D(
@@ -20,7 +12,7 @@ cdef extern from "unwrap_2d_ljmu.h":
             int image_width, int image_height,
             int wrap_around_x, int wrap_around_y,
             char use_seed, unsigned int seed
-            )
+            ) nogil
 
 
 def unwrap_2d(double[:, ::1] image,

--- a/skimage/restoration/_unwrap_3d.pyx
+++ b/skimage/restoration/_unwrap_3d.pyx
@@ -3,14 +3,6 @@
 # cython: nonecheck=False
 # cython: wraparound=False
 
-cdef extern from *:
-  void unwrap3D(double* wrapped_volume,
-                double* unwrapped_volume,
-                unsigned char* input_mask,
-                int image_width, int image_height, int volume_depth,
-                int wrap_around_x, int wrap_around_y, int wrap_around_z,
-                char use_seed, unsigned int seed) nogil
-
 
 cdef extern from "unwrap_3d_ljmu.h":
     void unwrap3D(
@@ -20,7 +12,7 @@ cdef extern from "unwrap_3d_ljmu.h":
             int volume_width, int volume_height, int volume_depth,
             int wrap_around_x, int wrap_around_y, int wrap_around_z,
             char use_seed, unsigned int seed
-            )
+            ) nogil
 
 
 

--- a/skimage/restoration/_unwrap_3d.pyx
+++ b/skimage/restoration/_unwrap_3d.pyx
@@ -11,6 +11,19 @@ cdef extern from *:
                 int wrap_around_x, int wrap_around_y, int wrap_around_z,
                 char use_seed, unsigned int seed) nogil
 
+
+cdef extern from "unwrap_3d_ljmu.h":
+    void unwrap3D(
+            double *wrapped_volume,
+            double *unwrapped_volume,
+            unsigned char *input_mask,
+            int volume_width, int volume_height, int volume_depth,
+            int wrap_around_x, int wrap_around_y, int wrap_around_z,
+            char use_seed, unsigned int seed
+            )
+
+
+
 def unwrap_3d(double[:, :, ::1] image,
               unsigned char[:, :, ::1] mask,
               double[:, :, ::1] unwrapped_image,

--- a/skimage/restoration/unwrap_2d_ljmu.c
+++ b/skimage/restoration/unwrap_2d_ljmu.c
@@ -39,6 +39,18 @@
 #define NOMASK 0
 #define MASK 1
 
+
+cdef extern from "unwrap_2d_ljmu.h":
+    void unwrap2D(
+            double *wrapped_image,
+            double *UnwrappedImage,
+            unsigned char *input_mask,
+            int image_width, int image_height,
+            int wrap_around_x, int wrap_around_y,
+            char use_seed, unsigned int seed
+            );
+
+
 typedef struct {
   double mod;
   int x_connectivity;

--- a/skimage/restoration/unwrap_2d_ljmu.c
+++ b/skimage/restoration/unwrap_2d_ljmu.c
@@ -40,17 +40,6 @@
 #define MASK 1
 
 
-cdef extern from "unwrap_2d_ljmu.h":
-    void unwrap2D(
-            double *wrapped_image,
-            double *UnwrappedImage,
-            unsigned char *input_mask,
-            int image_width, int image_height,
-            int wrap_around_x, int wrap_around_y,
-            char use_seed, unsigned int seed
-            );
-
-
 typedef struct {
   double mod;
   int x_connectivity;

--- a/skimage/restoration/unwrap_2d_ljmu.h
+++ b/skimage/restoration/unwrap_2d_ljmu.h
@@ -1,0 +1,9 @@
+void unwrap2D(
+        double *wrapped_image,
+        double *UnwrappedImage,
+        unsigned char *input_mask,
+        int image_width, int image_height,
+        int wrap_around_x, int wrap_around_y,
+        char use_seed, unsigned int seed
+        );
+

--- a/skimage/restoration/unwrap_3d_ljmu.c
+++ b/skimage/restoration/unwrap_3d_ljmu.c
@@ -40,6 +40,18 @@
 #define NOMASK 0
 #define MASK 1
 
+
+cdef extern from unwrap_3d_ljmu.h:
+    void unwrap3D(
+            double *wrapped_volume,
+            double *unwrapped_volume,
+            unsigned char *input_mask,
+            int volume_width, int volume_height, int volume_depth,
+            int wrap_around_x, int wrap_around_y, int wrap_around_z,
+            char use_seed, unsigned int seed
+            );
+
+
 typedef struct {
   double mod;
   int x_connectivity;

--- a/skimage/restoration/unwrap_3d_ljmu.c
+++ b/skimage/restoration/unwrap_3d_ljmu.c
@@ -41,16 +41,6 @@
 #define MASK 1
 
 
-cdef extern from unwrap_3d_ljmu.h:
-    void unwrap3D(
-            double *wrapped_volume,
-            double *unwrapped_volume,
-            unsigned char *input_mask,
-            int volume_width, int volume_height, int volume_depth,
-            int wrap_around_x, int wrap_around_y, int wrap_around_z,
-            char use_seed, unsigned int seed
-            );
-
 
 typedef struct {
   double mod;

--- a/skimage/restoration/unwrap_3d_ljmu.h
+++ b/skimage/restoration/unwrap_3d_ljmu.h
@@ -1,0 +1,9 @@
+void unwrap3D(
+        double *wrapped_volume,
+        double *unwrapped_volume,
+        unsigned char *input_mask,
+        int volume_width, int volume_height, int volume_depth,
+        int wrap_around_x, int wrap_around_y, int wrap_around_z,
+        char use_seed, unsigned int seed
+        );
+

